web-admin/src/components/projects/dashboards.ts

```diff
@@ -72,3 +78,66 @@ export function getDashboardListItemsFromFilesAndCatalogEntries(
 
   return dashboardListings;
 }
+
+export function useDashboardListItems(
+  instanceId: string,
+  project: CreateQueryResult<V1GetProjectResponse>
+): Readable<{
+  items: DashboardListItem[];
+  success: boolean;
+}> {
+  let isProfiling = false;
+  if (project) {
+    const status = get(project)?.data?.prodDeployment?.status;
+    if (
+      status === "DEPLOYMENT_STATUS_PENDING" ||
+      status === "DEPLOYMENT_STATUS_RECONCILING"
+    ) {
+      isProfiling = true;
+    }
+  }
+
+  return derived(
+    [
+      createRuntimeServiceListFiles(
+        instanceId,
+        {
+          glob: "dashboards/*.yaml",
+        },
+        {
+          query: {
+            placeholderData: undefined,
+            enabled: !isProfiling && !!project && !!instanceId,
+          },
+        }
+      ),
+      createRuntimeServiceListCatalogEntries(
+        instanceId,
+        {
+          type: "OBJECT_TYPE_METRICS_VIEW",
+        },
+        {
+          query: {
+            placeholderData: undefined,
+            enabled: !isProfiling && !!project && !!instanceId,
+          },
+        }
+      ),
+    ],
+    ([dashboardFiles, dashboardCatalogEntries]) => {
+      if (!dashboardFiles.isSuccess || !dashboardCatalogEntries.isSuccess)
+        return {
+          success: false,
```

Comment:
It'd be good to mirror the TanStack return types as much as possible. So `isSuccess` rather than `success`.